physicalplan/exchange/cancellable.go

```diff
@@ -11,6 +11,9 @@ import (
 	"github.com/thanos-community/promql-engine/physicalplan/model"
 )
 
+// TODO(bwplotka): Consider removing this and ensuring all operators check for context. It creates
```

Comment:
The idea was to make the planner inject this in the future. But we can also handle cancellations in each operator.

Reply:
Inject how?

Reply:
For example, the planner can either traverse the tree once more and wrap all nodes with cancellable. Anyway, let's leave this TODO now and we can decide later what to do. It can also be sufficient to handle context cancellation just in the engine before calling next.

Reply:
that  was my plan 💪🏽